onnxruntime/core/providers/webgpu/shader_variable.h

```diff
@@ -189,6 +189,10 @@ class ShaderVariableHelper : public ShaderIndicesHelper {
 
   friend class ShaderHelper;
 };
+#if defined(__GNUC__)
```

Comment:
Is this strict aliasing warning blocking the build? I think they are not (otherwise the CI should be already broken)

Since this PR is for implementing Softmax, I suggest to revert the unrelated changes. Changes like #23445 are doing similar fixes.